core/sdk/src/extn/extn_id.rs

```diff
@@ -140,6 +147,37 @@ pub struct ExtnId {
     pub service: String,
 }
 
+impl Serialize for ExtnId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_string())
+    }
+}
+
+impl<'de> Deserialize<'de> for ExtnId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        if let Ok(str) = String::deserialize(deserializer) {
+            if let Ok(id) = ExtnId::try_from(str) {
+                return Ok(id);
+            }
+        }
+        Err(serde::de::Error::unknown_variant("unknown", &["unknown"]))
+    }
+}
+
+impl PartialEq for ExtnId {
```

Comment:
Why do we need this ? Is there any difference compared to the default PartialEq implementation ?

Reply:
ExtnId will be used in the future for verifying if the request is coming from a specific Extension to return it back. PartialEq helps in validating the ExtnIds

Reply:
Yeah, my question was more about having a custom implementation here instead of using the default one.  Now we need to maintain this custom implementation. For instance, it will break if we add an additional member field to ExtnId Struct and forget to update this custom function with additional equality check.

Reply:
Agreed removed the PartialEq Implementation given all the fields now have partial eq capability